### PR TITLE
updated all dependencies and updated this to match said dependencies

### PR DESCRIPTION
**Pull request by Yuki-iso**
Not going to pretend to know what I'm doing, but running the bin after downloading it from cargo didn't work (due to outdated openCV version)

So I cloned, updated all dependencies to newest version, and fixed the errors indicated by `cargo c`

(`fast-image-resize` had some breaking changes, but it should all work exactly like before)

Never really do this so, hmu if I did something stupid :)

### CODE REVIEW

**Comment by maxcurzi** [comment]
What version of OpenCV did you use? And what OS? Currently I havent bumped up the required OpenCV version (from 4 to 6+) due to Ubuntu 22.04LTS

**Comment by maxcurzi** [comment]
All these dependencies are a bit of a headache. I knew this would happen some day :D 

I'm on KDE Neon which is in 22.04 and still wasn't rebased to Ubuntu 24.04 and ffmpeg is stuck at version 4.4.

As soon as I'll be able to upgrade OS, and assuming my 8-year old laptop with a NVIDIA card will happily accept the change (spoiler alert, it probably won't) I'll update this crate's various dependencies and every else with it.

Until then, any dependency version update will likely break something somewhere on someone's setup

**Comment by neon-mmd** [comment]
> All these dependencies are a bit of a headache. I knew this would happen some day :D
> 
> I'm on KDE Neon which is in 22.04 and still wasn't rebased to Ubuntu 24.04 and ffmpeg is stuck at version 4.4.
> 
> As soon as I'll be able to upgrade OS, and assuming my 8-year old laptop with a NVIDIA card will happily accept the change (it probably won't) I'll update this crate's various dependencies and every else with it.
> 
> Until then, any dependency version update will likely break something somewhere on someone's setup

I think one option to support opencv on all Linux distros would be that if we could compile our own version of opencv with tplay. For this there has already been an issue resolved earlier on `opencv` github repository I would suggest checking this out:

https://github.com/twistedfall/opencv-rust/issues/457

Or if the `opencv` maintainer accepts providing vendored feature support in the crate then that would be another way to have consitent opencv package installed on all Linux platforms. For this I would suggest maybe we can file a feature request about this on their repository and see if they accept it or not. :slightly_smiling_face:

File: Cargo.toml [hunk 1]
```diff
@@ -28,30 +28,30 @@ default-run = "tplay"
...
 thiserror = "1.0"
-fast_image_resize = "2.7"
-youtube_dl = "0.8"
+fast_image_resize = "4.2"
+youtube_dl = "0.10"
 tempfile = "3.5"
 url = "2.3"
 serde_json = "1.0"
-ffmpeg-next = "6.0"
+ffmpeg-next = "7.0.4"
```
Comment (nuffin) [inline Cargo.toml:47]: not work on ubuntu 22.04

**Pull request merged**